skipruntime-ts/adapters/postgres/src/index.ts

```diff
@@ -77,15 +101,39 @@ function validateKeyParam(params: Json): {
       };
       break;
     case "SERIAL":
+    case "SERIAL4":
     case "INTEGER":
+    case "INT":
+    case "INT4":
       select = (table: string, key: string) => {
         const keyNum = Number(key);
         // Checks if key is a safe 32-bit integer (and positive, if serial)
         if (
           !Number.isSafeInteger(keyNum) ||
           keyNum < min32bitInt ||
           keyNum > max32bitInt ||
-          (type == "SERIAL" && keyNum < 1)
+          (keyNum < 1 && ["SERIAL", "SERIAL4"].includes(type))
+        )
+          throw new SkipPostgresError(`Invalid ${type} key: ${key}`);
+        return format(
+          "SELECT * FROM %I WHERE %I = " + keyNum.toString() + ";",
+          table,
+          col,
+        );
+      };
+      break;
+    case "SMALLSERIAL":
```

Comment:
Tempted to suggest a factorization of all numeric cases...

Reply:
twist my arm! 28cbb7da366da27be519b06516d32d6472faf414